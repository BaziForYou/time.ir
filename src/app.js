// @flow
import express from "express";
import bodyParser from "body-parser";
import compress from "compression";
import helmet from "helmet";
import routes from "./routes";
import { notFound, errResponse, logRequest } from "./middlewares";

const app = express();

app
  .use(
    helmet({
      dnsPrefetchControl: false,
      frameguard: false,
      noCache: true
    })
  )
  .use(compress())
  .use(bodyParser.json())
  .use(bodyParser.urlencoded({ extended: true }))
  .use(logRequest)
  .use("/", routes)
  .use(notFound)
  .use(errResponse);

module.exports = app;
