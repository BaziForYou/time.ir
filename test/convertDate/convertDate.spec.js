// @flow

import request from "supertest";
import flatten from "flat";

import app from "../../src/app";

describe("GET /convert", () => {
  it(
    "returns a valid result on a valid request for hijri type",
    () => {
      return request(app)
        .get("/convert?year=1397&month=7&day=19&type=hijri")
        .expect(200)
        .then(res => {
          expect(res.body).toMatchSnapshot();
          const responseKeys = Object.keys(flatten(res.body));
          expect(responseKeys).toEqual([
            "jalali.numeral",
            "jalali.text",
            "gregorian.numeral",
            "gregorian.text",
            "hijri.numeral",
            "hijri.text"
          ]);
        });
    },
    20000
  );

  it(
    "returns a valid result on a valid request for jalali type",
    () => {
      return request(app)
        .get("/convert?year=1396&month=2&day=29&type=jalali")
        .expect(200)
        .then(res => {
          expect(res.body).toMatchSnapshot();
          const responseKeys = Object.keys(flatten(res.body));
          expect(responseKeys).toEqual([
            "jalali.numeral",
            "jalali.text",
            "gregorian.numeral",
            "gregorian.text",
            "hijri.numeral",
            "hijri.text"
          ]);
        });
    },
    20000
  );

  it(
    "returns a valid result on a valid request for gregorian type",
    () => {
      return request(app)
        .get("/convert?year=1394&month=9&day=14&type=gregorian")
        .expect(200)
        .then(res => {
          expect(res.body).toMatchSnapshot();
          const responseKeys = Object.keys(flatten(res.body));
          expect(responseKeys).toEqual([
            "jalali.numeral",
            "jalali.text",
            "gregorian.numeral",
            "gregorian.text",
            "hijri.numeral",
            "hijri.text"
          ]);
        });
    },
    20000
  );
});
