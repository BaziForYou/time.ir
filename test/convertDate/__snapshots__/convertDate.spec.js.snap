// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /convert returns a valid result on a valid request for gregorian type 1`] = `
Object {
  "gregorian": Object {
    "numeral": "1974-10-01",
    "text": "Tuesday, October 1, 1974",
  },
  "hijri": Object {
    "numeral": "1394-09-14",
    "text": "الثلاثاء، 14 رمضان 1394",
  },
  "jalali": Object {
    "numeral": "1353-07-09",
    "text": "سه شنبه، 9 مهر 1353",
  },
}
`;

exports[`GET /convert returns a valid result on a valid request for hijri type 1`] = `
Object {
  "gregorian": Object {
    "numeral": "2018-10-11",
    "text": "Thursday, October 11, 2018",
  },
  "hijri": Object {
    "numeral": "1440-02-01",
    "text": "الخمیس‬، 1 صفر 1440",
  },
  "jalali": Object {
    "numeral": "1397-07-19",
    "text": "پنج شنبه، 19 مهر 1397",
  },
}
`;

exports[`GET /convert returns a valid result on a valid request for jalali type 1`] = `
Object {
  "gregorian": Object {
    "numeral": "2017-05-19",
    "text": "Friday, May 19, 2017",
  },
  "hijri": Object {
    "numeral": "1438-08-22",
    "text": "الجمعة، 22 شعبان 1438",
  },
  "jalali": Object {
    "numeral": "1396-02-29",
    "text": "جمعه، 29 اردیبهشت 1396",
  },
}
`;
